bin/dev/run-tests-from-scratch

```
@@ -219,6 +218,16 @@ if [ -e "${SHARK_PROJ_DIR}/conf/shark-env.sh" ] ; then
 fi
 
 ####################################################################
+# Download Scala if SCALA_HOME is not specified.
+####################################################################
+if [ "x$SCALA_HOME" == "x" ] ; then
+  wget $SCALA_DOWNLOAD_PATH
```

hmm, yeah, you're right. It seemed a little dirty at first to use that -e flag and i thought we might want to stop using it at some point, but after thinking about it a little more, I think it actually might be the right thing to use in this script to avoid having ugly checks of $? all the time (like what I was suggesting here) since we really do want the fail fast behavior throughout most of this type of script.